### PR TITLE
Priority sorting doesn't work when set to DESC

### LINKED ISSUE #1768: Priority sorting doesn't work when set to DESC
Open any project and add &sort=priority%3Adesc to url to reproduce

### PR DESCRIPTION
**Pull request by DenisVorop**
## PR includes
- [x] Bug Fix

## Related issues
Resolve #1768 

## QA Instructions, Screenshots, Recordings

https://github.com/taskany-inc/issues/assets/95316053/7ad0d234-c82a-4163-92bd-426660865d94

### CODE REVIEW

**Review by 9teen90nine — APPROVED** [review]

File: src/components/GlobalSearch/GlobalSearch.tsx [hunk 1]
```diff
@@ -162,7 +162,7 @@ export const GlobalSearch = () => {
                                                     gap={10}
                                                     item={{
                                                         title: item.title,
-                                                        priority: item.priority,
+                                                        priority: item.priority?.title,
```
Comment (awinogradov) [inline src/components/GlobalSearch/GlobalSearch.tsx:165]: i18n?

File: src/components/GoalForm/GoalForm.tsx [hunk 2]
```diff
@@ -50,7 +49,7 @@ interface GoalFormProps extends React.HTMLAttributes<HTMLDivElement> {
     parent?: { id: string; title: string; flowId: string; description?: string | null };
     tags?: TagModel[];
     state?: State;
-    priority?: Priority | string;
+    priority?: PriorityReturnType;
```
Comment (awinogradov) [inline src/components/GoalForm/GoalForm.tsx:52]: Look at estimate type below. Let's do same thing ;)

File: src/components/GoalForm/GoalForm.tsx [hunk 3]
```diff
@@ -60,9 +59,15 @@ interface GoalFormProps extends React.HTMLAttributes<HTMLDivElement> {
     id?: string;
     tip?: React.ReactNode;
 
-    onSumbit: (fields: z.infer<GoalFormProps['validitySchema']>) => void;
+    onSubmit: (fields: z.infer<GoalFormProps['validitySchema']>) => void;
 }
 
+const defaultPriority = {
```
Comment (awinogradov) [inline src/components/GoalForm/GoalForm.tsx:65]: No way! Let's do that via req. Check flow dropdown for ex. Also add comment with issue which @9teen90nine mentioned.

File: trpc/queries/goals.ts [hunk 4]
```diff
@@ -89,7 +89,13 @@ export const goalsFilter = (
     where: Prisma.GoalFindManyArgs['where'];
     orderBy: Prisma.GoalFindManyArgs['orderBy'];
 } => {
-    const priorityFilter = data.priority?.length ? { priority: { in: data.priority } } : {};
+    const priorityFilter = data.priority?.length
+        ? {
+              priority: {
+                  title: { in: data.priority },
```
Comment (awinogradov) [inline trpc/queries/goals.ts:95]: Nope! You must pass priorityId here.

File: trpc/router/goal.ts [hunk 5]
```diff
@@ -346,6 +346,13 @@ export const goal = router({
 
             if (!actualGoal) return null;
 
+            const [previousPriority, actualPriority] = await Promise.all([
```
Comment (awinogradov) [inline trpc/router/goal.ts:349]: whats happening here?

File: trpc/router/goal.ts [hunk 6]
```diff
@@ -393,15 +400,15 @@ export const goal = router({
...
             }
 
-            if (actualGoal.priority !== input.priority) {
+            if (previousPriority && actualPriority && previousPriority?.id !== actualPriority?.id) {
                 history.push({
                     subject: 'priority',
                     action: 'change',
-                    previousValue: actualGoal.priority,
-                    nextValue: input.priority,
+                    previousValue: previousPriority.title,
```
Comment (awinogradov) [inline trpc/router/goal.ts:407]: Why not priorityId?

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/GlobalSearch/GlobalSearch.tsx (hunk 1 above)
Comment (DenisVorop) [reply src/components/GlobalSearch/GlobalSearch.tsx:165]: GoalListItemCompact uses i18n

File: trpc/router/goal.ts (hunk 6 above)
Comment (DenisVorop) [reply trpc/router/goal.ts:407]: in GoalActivity we pass strings to show what has changed to what

File: trpc/router/goal.ts [hunk 7]
```diff
@@ -393,15 +394,15 @@ export const goal = router({
...
             }
 
-            if (actualGoal.priority !== input.priority) {
+            if (actualGoal.priority?.id !== input.priority.id) {
                 history.push({
                     subject: 'priority',
                     action: 'change',
-                    previousValue: actualGoal.priority,
-                    nextValue: input.priority,
+                    previousValue: actualGoal.priority?.title ?? null,
```
Comment (awinogradov) [inline trpc/router/goal.ts:401]: U must push priority id here

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/PriorityFilter.tsx [hunk 8]
```diff
@@ -1,40 +1,38 @@
...
+interface PriorityFilterProps {
     text: string;
     value?: string[];
+    priorities: Priority[];
     onChange: (value: string[]) => void;
-}> = ({ value = [], onChange, text }) => {
+}
+
+function getKey(item: Priority) {
+    return item.id.toString();
```
Comment (awinogradov) [inline src/components/PriorityFilter.tsx:23]: It is String already

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Review by awinogradov — APPROVED** [review]

**Pull request merged**